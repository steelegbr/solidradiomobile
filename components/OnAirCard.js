/**
    Solid Radio Mobile App
    Copyright (C) 2020-2021 Marc Steele

    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.

    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with this program.  If not, see <https://www.gnu.org/licenses/>.
*/

import React from "react";
import { Component } from "react";
import { connect } from "react-redux";
import { Card, Paragraph, Title, Headline } from "react-native-paper";
import { generateStationTheme } from "../branding/branding";

/**
 * Card for displaying the currently on air show.
 */

class OnAirCard extends Component {

    render() {

        // Read in the properties

        const { onAir, theme } = this.props;

        // Render the card

        return (
            <Card theme={theme}>
                <Card.Cover source={{ uri: onAir.image }} />
                <Card.Content>
                    <Title>Now On Air - {onAir.show}</Title>
                    <Paragraph>{onAir.description}</Paragraph>
                </Card.Content>
            </Card>
        );

    }

}

function mapStateToProps(state, ownProps) {

    return {
        onAir: ownProps.onAir,
        theme: generateStationTheme(state.settings.darkMode, ownProps.station)
    };

}

export default connect(mapStateToProps)(OnAirCard);