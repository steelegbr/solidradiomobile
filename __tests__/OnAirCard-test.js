/**
 * Tests the on air carousel card.
 */

import React from 'react';
import OnAirCard from '../components/OnAirCard';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

// Redux

const mockStore = configureStore([]);
let store;

beforeEach(() => {
    store = mockStore({
        settings: {
            darkMode: false
        }
    });
});

// Tests

it('renders', () => {

    const onAir = {
        show: "Test Show",
        description: "A really good test description.",
        image: "art.png"
    };

    const station = {
        primary_colour: "#ff0000"
    };

    const onAirCard = renderer.create(
        <Provider store={store}>
            <OnAirCard onAir={onAir} station={station} />
        </Provider>
    ).toJSON();
    expect(onAirCard).toMatchSnapshot();

});
