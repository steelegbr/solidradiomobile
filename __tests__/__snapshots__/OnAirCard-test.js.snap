// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderRadius": 10,
      "elevation": 1,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 0.5,
        "width": 0,
      },
      "shadowOpacity": 0.24,
      "shadowRadius": 0.75,
    }
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flexGrow": 1,
        "flexShrink": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#eeeeee",
            "height": 195,
            "overflow": "hidden",
          },
          Object {
            "borderTopLeftRadius": 4,
            "borderTopRightRadius": 4,
          },
          undefined,
        ]
      }
    >
      <Image
        siblings={
          Array [
            "withTheme(Card.Cover)",
            "Card.Content",
          ]
        }
        source={
          Object {
            "uri": "art.png",
          }
        }
        style={
          Array [
            Object {
              "flex": 1,
              "height": undefined,
              "justifyContent": "flex-end",
              "padding": 16,
              "resizeMode": "cover",
              "width": undefined,
            },
            Object {
              "borderTopLeftRadius": 4,
              "borderTopRightRadius": 4,
            },
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "paddingHorizontal": 16,
          },
          Object {
            "paddingBottom": 16,
          },
          undefined,
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
            },
            Array [
              Object {
                "color": "rgba(0, 0, 0, 0.87)",
                "fontFamily": "System",
                "fontWeight": "500",
                "textAlign": "left",
                "writingDirection": "ltr",
              },
              Array [
                Object {
                  "fontSize": 20,
                  "letterSpacing": 0.15,
                  "lineHeight": 30,
                  "marginVertical": 2,
                },
                undefined,
              ],
              Array [
                Object {
                  "fontSize": 20,
                  "letterSpacing": 0.15,
                  "lineHeight": 30,
                  "marginVertical": 2,
                },
                undefined,
              ],
            ],
          ]
        }
      >
        Now On Air - 
        Test Show
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
            },
            Array [
              Object {
                "color": "rgba(0, 0, 0, 0.87)",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
                "writingDirection": "ltr",
              },
              Array [
                Object {
                  "fontSize": 14,
                  "letterSpacing": 0.25,
                  "lineHeight": 20,
                  "marginVertical": 2,
                },
                undefined,
              ],
              Array [
                Object {
                  "fontSize": 14,
                  "letterSpacing": 0.25,
                  "lineHeight": 20,
                  "marginVertical": 2,
                },
                undefined,
              ],
            ],
          ]
        }
      >
        A really good test description.
      </Text>
    </View>
  </View>
</View>
`;
